EventFilter/HcalRawToDigi/plugins/PackerHelp.h

```diff
@@ -587,4 +618,44 @@ class UHTRpacker {
   };
 };
 
+// converts HE QIE digies to HB data format
+
+QIE11DataFrame convertHB(QIE11DataFrame qiehe) {
+  QIE11DataFrame qiehb = qiehe;
+  int adc, tdc, capid;
+  bool soi;
+  int is = 0;
+
+  //  flavor for HB digies is hardcoded here
+  static const int hbflavor = 3;
+
+  //  iterator over samples
+  for (edm::DataFrame::const_iterator it = qiehe.begin(); it != qiehe.end(); ++it) {
+    if (it == qiehe.begin())
+      continue;
+    adc = qiehe[is].adc();
+    tdc = qiehe[is].tdc();
+    soi = qiehe[is].soi();
+
+    if (tdc >= 0 && tdc <= 24)
+      tdc = 0;
+    else if (tdc >= 25 && tdc <= 49)
+      tdc = 1;
+    else if (tdc == 63)
+      tdc = 2;
+    else if (tdc == 62)
```

Comment:
Aren't other values possible for ```qiehe[is].tdc()``` other than:
- 0<=tdc<=49
- 62
- 63 ?

(Even in case of errors, misconfigurations, etc...)

Reply:
According to the original QIE11 specifications, 0-49 for time information, 50-64 for special codes such as modes, lock, errors and special current condition. But HB has only 2 bits available for TDC, not all these information can be stored, so only the ones interested for physics are selected here. Special code 62 corresponds to the case when the current discriminator starts high. Special code 63 is when the TDC current discriminator starts low and remains low (nothing happened). @abdoulline I thought MC simulation is an ideal case where we don't have  such misconfiguration errors?

Reply:
I wonder whether it is ever possible that ```QIE11DataFrame::setSample(is, adc, tdc, soi)``` is called with a value of ```tdc``` different from 0-3, as it could be the case if codes 50-64 are returned, and what would happen in that case.
If there is not such a possibility the current implementation is fine, although it is hard to understand it by just looking at the code (and it may be error prone, I'd say...)

Reply:
This convertHB method is currently only used to convert MC QIE11, where only 0-49, 62 and 63 are assigned to the tdc, with their meanings mentioned above.